daemon/mgr/spec_namespace.go

```diff
@@ -2,39 +2,148 @@ package mgr
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	specs "github.com/opencontainers/runtime-spec/specs-go"
 )
 
+// isEmpty indicates whether namespace mode is empty.
+func isEmpty(mode string) bool {
+	return mode == ""
+}
+
+// isNone indicates whether container's namespace mode is set to "none".
+func isNone(mode string) bool {
+	return mode == "none"
+}
+
+// isHost indicates whether the container shares the host's corresponding namespace.
+func isHost(mode string) bool {
+	return mode == "host"
+}
+
+// isShareable indicates whether the containers namespace can be shared with another container.
+func isShareable(mode string) bool {
+	return mode == "shareable"
+}
+
+// isContainer indicates whether the container uses another container's corresponding namespace.
+func isContainer(mode string) bool {
```

Comment:
I think if string mode is `container:`, the return value is true as well.
Is this reasonable? @YaoZengzeng

Reply:
@allencloud yes I think it is OK, we should keep this judge function as simple as possible.

If the string is "container:", we won't get the corresponding container, it will still return an error in another place.

Reply:
What you mentioned is correct, currently we have not ruled defensive programming out. I think it is proper to set this rule at the entrance of API validation. Then no matter what changes in daemon side, we will not meet issues here or at other places.